buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchDistribution.java

```diff
@@ -0,0 +1,217 @@
+                "unknown platform [" + platform + "] for elasticsearch distribution [" + name + "], must be one of " + ALLOWED_PLATFORMS);
+        }
+        this.platform.set(platform);
+    }
+
+    public String getType() {
+        return type.get();
+    }
+
+    public void setType(String type) {
+        if (ALLOWED_TYPES.contains(type) == false) {
+            throw new IllegalArgumentException(
+                "unknown type [" + type+ "] for elasticsearch distribution [" + name + "], must be one of " + ALLOWED_TYPES);
+        }
+        this.type.set(type);
+    }
+
+    public String getFlavor() {
+        return flavor.get();
+    }
+
+    public void setFlavor(String flavor) {
+        if (ALLOWED_FLAVORS.contains(flavor) == false) {
+            throw new IllegalArgumentException(
+                "unknown flavor [" + flavor + "] for elasticsearch distribution [" + name + "], must be one of " + ALLOWED_FLAVORS);
+        }
+        this.flavor.set(flavor);
+    }
+
+    public boolean getBundledJdk() {
+        return bundledJdk.get();
+    }
+
+    public void setBundledJdk(boolean bundledJdk) {
+        this.bundledJdk.set(bundledJdk);
+    }
+
+    @Override
+    public String toString() {
+        return configuration.getSingleFile().toString();
+    }
+
+    public Extracted getExtracted() {
+        if (getType().equals("rpm") || getType().equals("deb")) {
+            throw new IllegalArgumentException("distribution type [" + getType() + "] for " +
+                "elasticsearch distribution [" + name + "] cannot be extracted");
+        }
+        return extracted;
+    }
+
+    @Override
+    public TaskDependency getBuildDependencies() {
+        return configuration.getBuildDependencies();
+    }
+
+    // internal, make this distribution's configuration unmodifiable
+    void finalizeValues() {
+
+        if (getType().equals("integ-test-zip")) {
+            if (platform.isPresent()) {
+                throw new IllegalArgumentException(
+                    "platform not allowed for elasticsearch distribution [" + name + "] of type [integ-test-zip]");
+            }
+            if (flavor.isPresent()) {
+                throw new IllegalArgumentException(
+                    "flavor not allowed for elasticsearch distribution [" + name + "] of type [integ-test-zip]");
+            }
+            if (bundledJdk.isPresent()) {
+                throw new IllegalArgumentException(
+                    "bundledJdk not allowed for elasticsearch distribution [" + name + "] of type [integ-test-zip]");
+            }
+            return;
+        }
+
+        if (getType().equals("archive")){
+            // defaults for archive, set here instead of via convention so integ-test-zip can verify they are not set
+            if (platform.isPresent() == false) {
+                platform.set(CURRENT_PLATFORM);
+            }
+        } else { // rpm or deb
+            if (platform.isPresent()) {
```

Comment:
Wouldn't specifying "linux" technically be ok even if it's superfluous?

Reply:
Yes it is technically ok. I'd rather keep it strict here, where for rpm/deb a platform is not specified.